build.gradle.kts

```diff
@@ -187,6 +187,7 @@ subprojects {
             allWarningsAsErrors = true
             jvmTarget = "1.8"
             apiVersion = "1.3"
+            freeCompilerArgs = freeCompilerArgs + "-Xallow-result-return-type"
```

Comment:
I didn't add this link because its first sentence says "`Result<T>` cannot be used as a direct result type of Kotlin functions" which is misleading because that's exactly what we do here, and also I've implemented you suggesting from below instead.